src/components/StorePage/StorePageContainer.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+
+// import AddStoreFormContainer from "../Forms/AddStore/AddStoreFormContainer";
+
+import { fetchStore } from "../../store/store/actions";
+
+export class StorePageContainer extends Component {
+  state = { toggle: false };
+
+  componentDidMount() {
+    const id = this.props.match.params.id;
+    this.props.fetchStore(id);
+  }
+
+  toggleAddForm = () => {
+    this.setState({
+      toggle: !this.state.toggle
+    });
+  };
+
+  render() {
+    const products =
+      this.props.store.products &&
+      this.props.store.products.length > 0 &&
+      this.props.store.products;
+
+    const store = this.props.store.store && this.props.store.store;
+
+    return (
+      <div>
+        {store && (
+          <div>
+            <p>name: {store.name}</p>
+            <p>address: {store.address}</p>
+            {products && (
+              <div>
+                {this.props.store.products &&
```

Comment:
you don't need this check do you? you already checked products exist and you have it in the `products` variable, so .map on products straight away